File: 01_find a max consecutive value in array/myalgo.c
#include <stdio.h>
#include <malloc.h>
#include <stdlib.h>
#include <time.h>

int* array;

void gen_nums(int n)
{
    srand(time(NULL));
    int i;
    for (i = 0; i < n; i++)
    {
        array[i] = rand() % 50 - 25;
    }
}

int main(int argc, char** argv)
{
    clock_t begin, end;
    int n;
    int max = 0;
    
    n = atoi(argv[1]);
    array = (int*)malloc(sizeof(int) * n);
  
    gen_nums(n);

    begin = clock();

    //* <------------------ [나만의 알고리즘]
    max = array[0];
    for (int m = 0; m < n;m++) {
        if (array[m]>max+array[m]) {
            max = array[m];
        }
        else {
            max = array[m]+max;
        }
    }
    //* ------------------------------------>

    end = clock();
    printf("max:%d\n",max);
    printf("Elapsed time: %f\n", (double)(end - begin) / CLOCKS_PER_SEC);
    return 1;
}

File: find a max consecutive value in array/myalgo.py
import sys
import time
from random import *

n = int(sys.argv[1])

array= []

sum = 0

for i in range(n):
    array.append(randrange(-25, 25))

begin = time.time()

# <--------------- 나만의 알고리즘

sum = array[0]
for i in range(n):
    sum = max(array[i], sum+array[i])

# ------------------------------->

end = time.time()
print(f'최대 부분합= {sum}')
print(f'Elapsed time: {end-begin:.7f} sec')
